overlord/snapstate/snapmgr.go

```diff
@@ -227,23 +260,42 @@ func Manager(s *state.State) (*SnapManager, error) {
 	return m, nil
 }
 
-// Store returns the store service used by the manager.
-func (m *SnapManager) Store() StoreService {
-	return m.store
+// ReplaceStore replaces the store used by the manager.
+func ReplaceStore(store StoreService) {
+	ubuntuStore = store
 }
 
-// ReplaceStore replaces the store used by manager.
-func (m *SnapManager) ReplaceStore(store StoreService) {
-	m.store = store
+var ubuntuStore StoreService
+
+// Store returns the snapstate store
+func Store() StoreService {
```

Comment:
as I put in a TODO comment a while ago I would make it actually a Store(state) helper and use the cache mechanism on the state to do this